lib/elliptic/eddsa/index.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+var bn = require('bn.js');
+var hash = require('hash.js');
+var elliptic = require('../../elliptic');
+var utils = elliptic.utils;
+var assert = utils.assert;
+
+function parseBytes(bytes) {
+  return typeof bytes === 'string' ? utils.toArray(bytes, 'hex') :
+                                     bytes;
+}
+
+function EDDSA(curve) {
+  if (!(this instanceof EDDSA))
+    return new EDDSA(curve);
+
+  assert(typeof curve === 'string');
+  assert(curve === 'ed25519', 'only tested with ed25519 so far');
+  var curve = this.curve = elliptic.curves[curve].curve;
+  this.g = curve.g;
+  this.g.precompute(curve.n.bitLength() + 1);
+  this.encBytes = Math.ceil(curve.n.bitLength() / 8);
+  this.hash = hash.sha512;
+}
+
+module.exports = EDDSA;
+
+/**
+* @param {Array} message - message bytes
+* @param {Array|String} seed - secret seed bytes
+* @returns {bytes} - signature
+*/
+EDDSA.prototype.sign = function sign(message, secret) {
+  message = parseBytes(message);
+  var derived = (Array.isArray(secret) || typeof secret === 'string') ?
+                this.deriveFromSeed(secret) : secret;
+  var r = this.hashToInt(derived.suffix, message);
+  var R = this.encodePoint(this.g.mul(r.mod(this.curve.n)));
+  var s_ = this.hashToInt(R, derived.encoded.A, message).mul(derived.a);
+  var S = r.add(s_).mod(this.curve.n);
+  return R.concat(this.encodeInt(S));
+};
+
+function pointEq(a, b) {
+  return a.getX().cmp(b.getX()) === 0 &&
+         a.getY().cmp(b.getY()) === 0;
+}
+
+/**
+* @param {Array} sig - sig bytes
+* @param {Array} message - message bytes
+* @param {Array|String} A - public key bytes
+* @returns {bytes} - true if
+*/
+EDDSA.prototype.verify = function verify(sig, message, A) {
+  sig = parseBytes(sig);
+  message = parseBytes(message);
+  var R = sig.slice(0, this.encBytes);
+  var Ap = this.decodePoint(A);
+  var S = this.decodeRedInt(sig.slice(this.encBytes));
+  var h = this.hashToInt(R, A, message).mod(this.curve.n);
+  var SG = this.g.mul(S);
+  var RplusAh = this.decodePoint(R).add(Ap.mul(h));
+  return pointEq(RplusAh, SG);
+};
+
+/**
+* @returns {bn} little-endian intepretation of hash of one or more
+*               Arrays of bytes.
+*/
+EDDSA.prototype.hashToInt = function hashToInt() {
+  var hash = this.hash();
+  for (var i = 0; i < arguments.length; i++)
+    hash.update(arguments[i]);
+  return this.decodeInt(hash.digest());
+};
+
+EDDSA.prototype.deriveFromSeed = function deriveFromSeed(seed) {
+  seed = parseBytes(seed);
+  assert(seed.length === this.encBytes);
+
+  var hash = this.hash().update(seed).digest();
+  var lastIx = this.encBytes - 1;
+
+  var a = hash.slice(0, this.encBytes);
+  a[0] &= 248;
+  a[lastIx] &= 127;
+  a[lastIx] |= 64;
+
+  var aBn = this.decodeRedInt(a);
+  var A = this.g.mul(aBn);
+
+  return {
```

Comment:
That makes sense, do you want a full KeyPair class etc? I can try and match the api with EC?

It's 1:30am here, and I'm about to sleep. I'm more than happy if you form it as you'd like yourself if you are :) Otherwise tomorrow.